Client/Frontend/UserContent/UserScripts/Sandboxed/MainFrame/AtDocumentStart/AdblockRustInjector.js

```diff
@@ -0,0 +1,58 @@
+"use strict";
+
+Object.defineProperty(window.__firefox__, "AdblockRustInjector", {
+  enumerable: false,
+  configurable: false,
+  writable: false,
+  value: {}
+});
+
+Object.defineProperty(window.__firefox__.AdblockRustInjector, "applyFilter", {
+  enumerable: false,
+  configurable: false,
+  writable: false,
+  value: function(json) {
+      var head = document.head || document.getElementsByTagName('head')[0];
+      if (head == null) {
+          return;
+      }
+      
+      const obj = JSON.parse(atob(json));
+      const hide_selectors = obj.hide_selectors;
+      const style_selectors = obj.style_selectors;
+      const injected_script = obj.injected_script;
+      
+      // CSS Rules
+      var rules = "";
+      var style = document.createElement('style');
+      style.type = 'text/css';
+
+      for (const selector of hide_selectors) {
+          rules += selector + "{display: none !important}"
+      }
+      
+      for (const [key, value] of Object.entries(style_selectors)) {
+          const value = entry.value;
+          
+          var subRules = "";
+          for (const subRule of value) {
+              subRules += subRule + ";"
+          }
+          
+          rules += key + "{" + subRules + "}"
+      };
+
+      if (style.styleSheet) {
+        style.styleSheet.cssText = rules;
+      } else {
+        style.appendChild(document.createTextNode(rules));
+      }
+
+      head.appendChild(style);
+      
+      // Scripts
+      var script = document.createElement("script");
+      script.innerHTML = injected_script;
+      head.appendChild(script);
```

Comment:
I'm not familiar with iOS injection methods, but if we have an option of something more privileged than inserting `<style>` and `<script>` elements then we should look into it. For example, the stylesheets can be fairly easy for pages to workaround using `!important` styles.

If nothing more powerful is available then this is fine, but we might want to add `!important` to the rules from `style_selectors`.

Reply:
The only other way is to execute the JS directly and use `document.querySelector('....')`, but this can get slow.
Adding the script to the page is the only other alternative that I know of.